src/librustc_codegen_ironox/ir/function.rs

```diff
@@ -36,11 +36,12 @@ impl IronOxFunction {
         fn_type: Type) -> IronOxFunction {
         match cx.types.borrow()[fn_type] {
             LLType::FnType { ref args, ref ret } => {
-                let ret = Value::Param(*ret);
                 let mut params = Vec::with_capacity(args.len());
                 for (index, arg_ty) in args.iter().enumerate() {
-                    params.push(Value::Param(*arg_ty));
+                    params.push(Value::Param(index, *arg_ty));
                 }
+                 // FIXME: ret should not have an index...
```

Comment:
What does this mean? Can we put in a guard against this going wrong?

Reply:
`Value::Param(index, type)` is a value that represents the parameter of a function. The return value is not actually a parameter, but I am still representing it as `Value::Param` (it's just that its `index` is invalid). This is a limitation of my IR.

Reply:
OK. Is there an `assert` or some other check that we can put here?

Reply:
This is confusing, which means it was a bad idea to begin with. `ret` (the return type of the function) is now simply a `Type` (and not a `Param`).

A `Param` is just a wrapper around the `Type` of an argument. The arguments of a function are stored as a vector of `Value::Param`s. They are `Value`s (not `Type`s), because the backend needs to implement a `get_param(index)` function that returns parameters as a `Value`s.